### PR TITLE
Issue #5401 - Moving jetty-http-tools top level project.

### LINKED ISSUE #5401: Move jetty-http-tools under the project root
**Jetty version**
10.0.x

**Description**
`HttpTester` is a useful tool to test HTTP requests/responses.

In Jetty 9.4.x, this was available in `jetty-http` with the classifier `tests`.

In Jetty 10.0.x it's not deployed to Maven Central because it's under `$JETTY_ROOT/tests/`, but would be great if it is available again in Central for other projects such as CometD to use it.

### PR DESCRIPTION
**Pull request by joakime**
+ Ensures that it will be deployed.
+ Adding to generated (aggregate-like) javadoc as well.

Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: jetty-http-tools/src/main/java/module-info.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+module org.eclipse.jetty.http.tools
+{
+    exports org.eclipse.jetty.http.tools;
+    exports org.eclipse.jetty.http.tools.matchers;
+
+    requires transitive org.eclipse.jetty.http;
+
+    // Optional - only required if you use hamcrest too.
+    requires static org.hamcrest;
```
Comment (sbordet) [inline jetty-http-tools/src/main/java/module-info.java]: OMG. Just to support `assertThat()` now we need to depend on an automatic module.
Can we just get rid of the `hamcrest` dependency? It adds _nothing_ and just complicates things -- everything is already available in the `HttpFields` API.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-http-tools/src/main/java/module-info.java (hunk 1 above)
Comment (joakime) [reply jetty-http-tools/src/main/java/module-info.java]: I looked into that (getting rid of hamcrest in jetty-http-tools), but it made for a MUCH larger PR to change the existing code (in the rest of jetty) that uses these hamcrest features exposed in jetty-http-tools.
If you are gung-ho on this, I'll start work on removing it.

Comment (gregw) [reply jetty-http-tools/src/main/java/module-info.java]: I like the hamcrest style!

Comment (joakime) [reply jetty-http-tools/src/main/java/module-info.java]: Alright, will do.
I'll rebuild this PR from scratch with the jetty-http-tools split between top level and matchers still in /tests/

**Comment by joakime** [comment]
Will rebuild in cleaner way.

**Pull request closed without merging**